content/en/blog/2023/any-metric-receiver.md

```diff
@@ -0,0 +1,480 @@
+# prettier-ignore
+cspell:ignore: openssl servername noout enddate gomod debugexporter otlpexporter otlphttpexporter transformprocessor otlpreceiver carbonreceiver ottl datapoint webserver Helmuth
+---
+
+While OpenTelemetry (OTel) is here to help you with troubleshooting and handling
+the _"unknown unknowns"_, it is also instrumental for managing route tasks like
+monitoring system metrics, like disk usage, server availability or SSL
+certificate expiration dates. This can be achieved by utilizing any one of the
+[90+ receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver)
+available for the [OpenTelemetry Collector](/docs/collector), such as the
+[Host Metrics Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver)
+or the [HTTP Check Receiver](/blog/2023/synthetic-testing/).
+
+But what if the available receivers don't meet your specific needs? Suppose you
+have a collection of shell scripts that provide custom metrics, and you want to
+export these to the OpenTelemetry Collector. You could write your own receiver,
+but this would require proficiency in Go programming language and potentially
+converting your scripts to Go.
+
+Before embarking on this path, consider examining the available receivers more
+closely: Some of them are capable of assimilating metrics in different
+formats—like
+[Carbon](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/carbonreceiver),
+[StatsD](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/statsdreceiver),
+[InfluxDB](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/influxdbreceiver),
+[Prometheus](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver)
+and even
+[SNMP](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/snmpreceiver)—and
+integrating them into the OpenTelemetry ecosystem. With minor tweaks to your
+shell scripts, you can use one of these receivers to achieve your objective. For
+instance, the Carbon Receiver, with its simple
+[plaintext protocol](https://graphite.readthedocs.io/en/stable/feeding-carbon.html#the-plaintext-protocol),
+is ideal for use with shell scripts. Its protocol is incredibly straightforward:
+
+> The plaintext protocol is the most straightforward protocol supported by
+> Carbon. The data sent must be in the following format:
+> `<metric path> <metric value> <metric timestamp>`.
+
+## Example script: Check certificate expiration
+
+Consider the following shell script, which accepts a host name as an argument,
+and uses
+[`openssl s_client`](https://www.openssl.org/docs/manmaster/man1/openssl-s_client.html)
+to retrieve the certificate and compute the remaining time until certificate
+expiration:
+
+```shell
+#!/bin/bash
+URL=${1}
```

Comment:
How about using `HOST` rather than `URL` for the host name (here and elsewhere)?

```suggestion
HOST=${1}
```